bin/live/pycbc_live_single_trigger_fits

```diff
@@ -0,0 +1,377 @@
+    cut_params[nsidx] = 'snr'
+    nsineq = cut_inequalities[nsidx]
+    nsthresh = cut_thresholds[nsidx]
+
+cut_params = list(cut_params)
+cut_inequalities = list(cut_inequalities)
+cut_thresholds = list(cut_thresholds)
+
+# The following datasets are required for calculation of cut parameters
+required_datasets = ['snr', 'chisq', 'chisq_dof', 'template_duration',
+                     'mass1', 'mass2', 'spin1z', 'spin2z']
+required_datasets += cut_params
+required_datasets = list(set(required_datasets))
+
+# Make some strings for logging /saving setting for attributes
+zip_inequs = zip(cut_params, cut_inequalities, cut_thresholds)
+ineq_str = ''
+for p, i, v in zip_inequs:
+    ineq_str += '{} {} {}, '.format(p, i, v)
+ineq_str = ineq_str[:-2]
+
+all_zip_inequs = zip(args.cut_params, args.cut_inequalities,
+                     args.cut_thresholds)
+ineq_str_all = ''
+for p, i, v in all_zip_inequs:
+    ineq_str_all += '{} {} {}, '.format(p, i, v)
+ineq_str_all= ineq_str_all[:-2]
+logging.info('Criteria set up as ' + ineq_str_all)
+
+# Also calculate live time so that this fitting can be used in rate estimation
+# Live time is not immediately obvious - get an approximation with 8 second
+# granularity by adding 8 seconds per 'valid' file
+
+live_time = {ifo: 0 for ifo in args.ifos}
+
+#store as a DictArray - nice as it has the .select() method
+events_dict = {k : [] for k in required_datasets}
+logging.info("Getting events from which meet criteria: {}".format(ineq_str))
+
+# Loop through files - add events which meet the immediately gettable
+# criteria
+date_directory = os.path.join(args.top_directory, args.analysis_date)
+events = {i: DictArray(data={rp: np.array([]) for rp in required_datasets}) for i in args.ifos}
+files = [f for f in os.listdir(date_directory)
+         if args.file_identifier in f]
+counter = 0
+for filename in files:
+    counter += 1
+    if counter % 1000 == 0:
+        logging.info("Processed %d files" % counter)
+        for ifo in args.ifos:
+            logging.info("{}: {} triggers in {}s".format(ifo,
+                                                        events[ifo].data['snr'].size,
+                                                        live_time[ifo]))
+    f = os.path.join(date_directory, filename)
+    skipping_file = False
+    #If there is an IOerror with the file, don't fail, just carry on
+    try:
+        h5py.File(f,'r')
+    except IOError:
+        logging.info('IOError with file ' + f)
+        continue
+    with h5py.File(f,'r') as fin:
+        # Open the file: does it have the ifo group and snr dataset?
+        for ifo in args.ifos:
+            if not (ifo in fin and 'snr' in fin[ifo]):
+                continue
+            live_time[ifo] += 8
+            # all the required datasets must be present - this should
+            # not happen though as if snr is present then the others should be
+            if any([rd not in fin[ifo] for rd in required_datasets]):
```

Comment:
You can drop the square brackets; `any` can handle a generator and do short-circuit evaluation.

Reply:
Also a good thing to know - it's used elsewhere so I will fix those as well